api/models/datastore.go

```diff
@@ -59,6 +59,24 @@ type Datastore interface {
 	// ErrDatastoreEmptyRoutePath when routePath is empty. Returns ErrRoutesNotFound when no route exists.
 	RemoveRoute(ctx context.Context, appID, routePath string) error
 
+	// PutFn inserts a new function if one does not exist, applying any defaults necessary, or
+	// updates a function that exists under the same name. Returns ErrDatastoreEmptyFn if func is nil,
```

Comment:
Not a big fan of this idea. Can see it causing users a lot of trouble with people over writing others changes.